rcsb/workflow/wuw/PdbCsmImageWorkflow.py

```diff
@@ -0,0 +1,157 @@
+"""Images workflow functions that actually do the work."""
+
+import gzip
+import json
+import datetime
+import random
+from pathlib import Path
+import logging
+import subprocess
+
+logger = logging.getLogger(__name__)
+
+
+def get_pdb_list(pdb_gz_path: str, update_all_images: bool, pdb_base_dir: str) -> list:
+    """Build pdb list via pdb gz file."""
+    pdb_ids_timestamps = {}
+    with gzip.open(pdb_gz_path) as f:
+        data = json.loads(f.read())
+        for id_val in data:
+            datetime_object = datetime.datetime.strptime(data[id_val], "%Y-%m-%dT%H:%M:%S%z")
+            pdb_ids_timestamps[id_val.lower()] = datetime_object
+    pdb_id_list = []
+    if update_all_images:
+        for id_val in pdb_ids_timestamps:
+            path = id_val[1:3] + "/" + id_val + ".bcif"
+            pdb_id_list.append(f"{id_val} {path} experimental")
+    else:
+        for id_val, timestamp in pdb_ids_timestamps.items():
+            path = id_val[1:3] + "/" + id_val + ".bcif"
+            bcif_file = pdb_base_dir + path
+            if Path.exists(bcif_file):
+                t1 = Path.stat(bcif_file).st_mtime
+                t2 = timestamp.timestamp()
+                if t1 < t2:
+                    pdb_id_list.append(f"{id_val} {path} experimental")
+            else:
+                pdb_id_list.append(f"{id_val} {path} experimental")
+    return pdb_id_list
+
+
+def get_csm_list(csm_gz_path: str, update_all_images: bool, csm_base_dir: str) -> list:
+    """Build csm list via csm gz file."""
+    with gzip.open(csm_gz_path) as f:
+        data = json.loads(f.read())
+        dic = {}
+        for model_id in data:
+            item = data[model_id]
+            item["modelPath"] = item["modelPath"].lower()  # prod route of BinaryCIF wf produces lowercase filenames
+            item["datetime"] = datetime.datetime.strptime(item["lastModifiedDate"], "%Y-%m-%dT%H:%M:%S%z")
+            dic[model_id.lower()] = item
+    model_ids_metadata = dic
+    model_list = []
+
+    if update_all_images:
+        for model_id, metadata in model_ids_metadata.items():
+            model_path = metadata["modelPath"].replace(".cif", ".bcif").replace('.gz', '')
+            model_list.append(f"{model_id} {model_path} computational")
+    else:
+        # 'incremental' for weekly
+        for model_id, metadata in model_ids_metadata.items():
+            model_path = metadata["modelPath"].replace(".cif", ".bcif").replace('.gz', '')
+            bcif_file = csm_base_dir + model_path
+            if Path.exists(bcif_file):
+                t1 = Path.stat(bcif_file).st_mtime
+                t2 = metadata["datetime"].timestamp()
+                if t1 < t2:
+                    model_list.append(f"{model_id} {model_path} computational")
+            else:
+                model_list.append(f"{model_id} {model_path} computational")
+    return model_list
+
+
+def images_gen_lists(args: dict) -> None:
```

Comment:
If you need to provide a large dict of arguments (instead of specifying them each explicitly), I would recommend using `**kwargs` instead of an argument called `args`. E.g., see usage at https://github.com/rcsb/py-rcsb_exdb/blob/master/rcsb/exdb/wf/ExDbWorkflow.py#L76 (in which you then grab the arguments you need using `kwargs.get(...)`, which lets you also specify default values).

Reply:
@piehld I set the default values in the arg parse. Should I instead set them in .get() ?

Reply:
I don't think it hurts to have redundancy, so I would suggest leaving them in the arg parse but doing so again here.